Index: logrotate-3.8.5/logrotate.c
===================================================================
--- logrotate-3.8.5.orig/logrotate.c	2013-06-26 22:17:20.821656206 +0100
+++ logrotate-3.8.5/logrotate.c	2013-06-26 22:17:20.817656132 +0100
@@ -314,8 +314,12 @@
     if (fchmod(fd, (S_IRUSR | S_IWUSR) & sb->st_mode)) {
 	message(MESS_ERROR, "error setting mode of %s: %s\n",
 		fileName, strerror(errno));
-	close(fd);
-	return -1;
+	if (geteuid() == 0) {
+		/* If run as non-root, we may not be able to duplicate
+		   the ownership of the file. Warn but don't fail. */
+		close(fd);
+		return -1;
+	}
     }
 
 	if (fstat(fd, &sb_create)) {
