Index: logrotate-3.7.9/config.c
===================================================================
--- logrotate-3.7.9.orig/config.c	2011-06-22 15:40:42.727459027 +0100
+++ logrotate-3.7.9/config.c	2011-06-22 15:50:46.047374110 +0100
@@ -165,7 +165,7 @@
 static int checkFile(const char *fname)
 {
 	int i;
-	char pattern[PATH_MAX];
+	char *pattern;
 
 	/* Check if fname is '.' or '..'; if so, return false */
 	if (fname[0] == '.' && (!fname[1] || (fname[1] == '.' && !fname[2])))
@@ -173,15 +173,17 @@
 
 	/* Check if fname is ending in a taboo-extension; if so, return false */
 	for (i = 0; i < tabooCount; i++) {
-		snprintf(pattern, sizeof(pattern), "*%s", tabooExts[i]);
+		asprintf(&pattern, "*%s", tabooExts[i]);
 		if (!fnmatch(pattern, fname, 0))
 		{
 			message(MESS_DEBUG, "Ignoring %s, because of %s ending\n",
 					fname, tabooExts[i]);
+			free(pattern);
 			return 0;
 		}
 	}
 
+	free(pattern);
 	/* All checks have been passed; return true */
 	return 1;
 }
